### DISCUSSION TITLE
[MRTCore] How to load resources across multiple .pri files

### DISCUSSION THREAD

**Discussion by MarkIngramUK** [microsoft/WindowsAppSDK#1239] [Q&A] [answered]
The default config created from `MakePri` will create a packaging section as follows:

```
  <packaging>
    <autoResourcePackage qualifier="Language"/>
    <autoResourcePackage qualifier="Scale"/>
    <autoResourcePackage qualifier="DXFeatureLevel"/>
  </packaging>
```

This means that when I add translated strings to my app, the translations will exist in another pri file, e.g.

`myresources.language-de-DE.pri`

However, when you create a `ResourceManager` you can only provide it with a single filename, e.g.

`auto manager = factory.CreateInstance(L"myresources.pri");`

So, how do I load my German strings in this instance? Do I need multiple `ResourceManager` instances? This seems redundant given that I can request qualifier overloads such as:

```
auto context = manager.CreateResourceContext();
context.QualifierValues().Insert(L"Language", L"de-DE");
auto candidate = manager.MainResourceMap().GetValue(L"mystring", context);
```

What is the solution here? Previously, in WPF, we would have had satellite assemblies that contained the translated resources, so I'm assuming that we can do similar with MRTCore?

When I tried requesting the German resources from `myresources.pri` I ended up with only English (regardless of qualifier override). When I loaded `myresources.language-de-DE.pri` I ended up with only German strings.

**Comment by axelandrejs** [comment, marked as answer]
Splitting the resources by qualifier is to support resource packs. If you build your app as a bundle, the various resource files will be put into different packages, which are then deployed as needed depending on the device configuration. In that case, the deployment stack will create a merged view of all the resource files, and MRT core will load that transparently if you use the default constructor.

If you're not building a bundle, or if your app is unpackaged, then the right thing to do is to remove the packaging entries in the config. That way all the strings end up in the same file.

You can find more details here: https://docs.microsoft.com/en-us/windows/uwp/app-resources/build-resources-into-app-package
